### DISCUSSION TITLE
error:  Call to undefined method Phalcon\Paginator\Adapter\Model::getPaginate()

### DISCUSSION THREAD

**Discussion by vlad2005** [phalcon/cphalcon#15406] [Q&A] [answered]
Hi!
I try to use Paginator but i got this error:
`error:  Call to undefined method Phalcon\Paginator\Adapter\Model::getPaginate()`

My code is this (from documentation)
`
use Scmd\Test;
use Phalcon\Paginator\Adapter\Model as PaginatorModel;

class TestController extends ControllerBase
{

    public function indexAction()
    {
    	
		$currentPage = isset($_GET['page']) ? (int) $_GET['page'] : 1;

		$paginator = new PaginatorModel(
			[
				'data'  => Test::find(),
				'limit' => 5,
				'page'  => $currentPage,
			]
		);

		$page = $paginator->getPaginate();
    	        $this->view->items = $page;
    }
`
Thanks in advance!

**Comment by Jeckerson** [comment]
Hello, what is your PHP and Phalcon versions?

**Reply by vlad2005** [reply]
Phalcon 4.1.0
Php 7.4.16

**Comment by ekmst** [comment]
```
$paginator = new PaginatorModel(
		[
			'model'  => Test::class,
			'limit' => 5,
			'page'  => $currentPage,
		]
	);
```

**Reply by vlad2005** [reply]
Same error.

**Comment by vlad2005** [comment]
It seems that the `getPagination ()` method does not exist, although it is presented in the documentation. The documentation should be updated.
`getPagination ()` must be replaced by `pagination ()`

Example:
`$page = $paginator->getPaginate();` should be `$page = $paginator->paginate();`

**Reply by Ultimater** [reply]
In 3.4 the getPaginate method was deprecated but still available:
https://github.com/phalcon/cphalcon/blob/3.4.x/phalcon/paginator/adapter/model.zep
In 4.0 it was removed:
https://github.com/phalcon/cphalcon/blob/4.0.x/phalcon/Paginator/Adapter/Model.zep

As you found, you need to use the paginate() method instead.

When I check the tutorial, I see `paginate()` used.
https://docs.phalcon.io/4.0/en/tutorial-invo

Where in the documentation did you find a reference to `getPaginate()`?

**Reply by vlad2005** [reply]
Thank you!!
It's my fault, getPagination () does not appear in the documentation available for Phalcon 4.0. Most likely I looked at the 3.0 documentation.
My apologies!

**Comment by Ultimater** [comment, marked as answer]
I'm marking this as answered as it was most likely due to following an older documentation.
